ios/Handlers/ResourcesHandler.swift

```diff
@@ -1,137 +1,228 @@
+        guard let headers = details["headers"] as? [String: Any] else {
+            throw WrapperError.parametersNotCorrect(
+                description: "'headers' must be an object of containing String: String")
         }
+        return headers.compactMapValues {$0 as? String}
     }
     
-    fileprivate func anonymousResourcesRequest(_ details: NSDictionary, _ completion: @escaping (String?, Error?) -> Void) {
-        guard let path = details["path"] as? String else {
-            let error = NSError(domain: ONGFetchImplicitResourceErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey : "'path' can not be empty"])
-            completion(nil, error)
-            return
+    
+    // We copy over the swift api from the iOS SDK due to a bug in the sdk, because not all classes it uses are public we need to copy those over aswell. All below code can be removed once that fix is in the iOS SDK.
+    
+    @discardableResult
+    func sendRequest(_ resourceRequest: ResourceRequest, completion: @escaping ((_ response: ResourceResponse?, _ error: Error?) -> Void)) -> NetworkTask? {
+        let ongNetworkTask = ONGDeviceClient.sharedInstance().fetchResource(resourceRequest.ongRequest) { response, error in
+            completion(response != nil ? ResourceResponseImplementation(response!) : nil, error)
         }
-        guard let method = details["method"] as? String else {
-            let error = NSError(domain: ONGFetchImplicitResourceErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey : "'method' can not be empty"])
-            completion(nil, error)
-            return
+        guard let ongNetworkTask = ongNetworkTask else { return nil }
+        return NetworkTaskImplementation(ongNetworkTask)
+    }
+    
+    
+    
+    class ResourceResponseImplementation: ResourceResponse {
+        var response: HTTPURLResponse
+        var allHeaderFields: [AnyHashable : Any]
+        var statusCode: Int
+        var data: Data?
+        
+        init(_ resourceResponse: ONGResourceResponse) {
+            self.response = resourceResponse.rawResponse
+            self.allHeaderFields = resourceResponse.allHeaderFields
+            self.statusCode = resourceResponse.statusCode
+            self.data = resourceResponse.data
         }
-        let request = ONGResourceRequest(path: path, method: method, parameters: details["parameters"] as? [String : Any], encoding: ONGParametersEncoding.formURL, headers: details["headers"] as? [String : String]);
-
-        ONGDeviceClient.sharedInstance().fetchResource(request) { response, error in
-            if let error = error {
-                completion(nil, error)
+        
+        required init(response: HTTPURLResponse, data: Data?) {
+            self.response = response
+            self.allHeaderFields = response.allHeaderFields
+            self.statusCode = response.statusCode
+            self.data = data
+        }
+    }
+    
+    class NetworkTaskImplementation: NetworkTask {
```

Comment:
Is it used only in this file (copied from SDK?)? If so please mark it as `private`, otherwise move to separate file.